### PR TITLE
dev-python/logfury: Enable python 3.7 and 3.8

### PR DESCRIPTION
**Pull request by WPettersson**
Closes: https://bugs.gentoo.org/719090
Package-Manager: Portage-2.3.89, Repoman-2.3.21

### CODE REVIEW

**Comment by WPettersson** [comment]
The actual tests themselves are in the git repository for logfury (see https://github.com/ppolewicz/logfury), but the tests are not distributed via pypi. I don't know why, I didn't write the package, but this makes it hard (if not impossible?) to do testing within portage. If there are any ways around this, I'd be interested to hear them.

As for test results, they can be found on travis-ci at https://travis-ci.org/github/WPettersson/logfury/jobs/678809377 (python3.7) and https://travis-ci.org/github/WPettersson/logfury/jobs/678809378 (python3.8). I have also successfully run the tests in the git repository on my Gentoo box to ensure they all pass.

**Comment by mgorny** [comment]
That's entirely normal in hundreds of Python packages. Just fetch archive from GitHub and forget about pypi.

**Comment by WPettersson** [comment]
New commit force-pushed fetches from Github, and runs tests, which pass.

File: dev-python/logfury/logfury-0.1.2-r1.ebuild [hunk 1]
```diff
@@ -3,13 +3,14 @@
 
 EAPI=6
 
-PYTHON_COMPAT=( python3_6 )
+PYTHON_COMPAT=( python3_6 python3_7 python3_8 )
+DISTUTILS_USE_SETUPTOOLS=bdepend
```
Comment (mgorny) [inline dev-python/logfury/logfury-0.1.2-r1.ebuild:7]: That's the default.

File: dev-python/logfury/logfury-0.1.2-r1.ebuild [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
 
 inherit distutils-r1
 
 DESCRIPTION="Boilerplate library for logging method calls"
 HOMEPAGE="https://github.com/ppolewicz/logfury https://pypi.org/project/logfury/"
-SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+SRC_URI="https://github.com/ppolewicz/${PN}/archive/${PV}.tar.gz -> ${P}a.tar.gz"
```
Comment (mgorny) [inline dev-python/logfury/logfury-0.1.2-r1.ebuild:13]: Use `.gh.tar.gz` not to introduce false versions.

File: dev-python/logfury/logfury-0.1.2-r1.ebuild [hunk 3]
```diff
@@ -18,4 +19,7 @@ KEYWORDS="~amd64 ~x86"
 RDEPEND="dev-python/six[${PYTHON_USEDEP}]"
 
 DEPEND="${RDEPEND}
-	dev-python/setuptools[${PYTHON_USEDEP}]"
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/logfury/logfury-0.1.2-r1.ebuild:22]: This is redundant.

File: dev-python/logfury/logfury-0.1.2-r1.ebuild (hunk 1 above)
Comment (WPettersson) [reply dev-python/logfury/logfury-0.1.2-r1.ebuild]: Removed default.

File: dev-python/logfury/logfury-0.1.2-r1.ebuild (hunk 3 above)
Comment (WPettersson) [reply dev-python/logfury/logfury-0.1.2-r1.ebuild]: Removed redundant setuptools requirement.

File: dev-python/logfury/logfury-0.1.2-r1.ebuild (hunk 2 above)
Comment (WPettersson) [reply dev-python/logfury/logfury-0.1.2-r1.ebuild]: Now uses .gh instead of .a

**Comment by WPettersson** [comment]
Note that I also found two tests that use nose, but were not mentioning in the accompanying documentation. I've added that dependency to the ebuild

**Comment by mgorny** [comment]
You are missing some dependencies.

```
>>> Test phase: dev-python/logfury-0.1.2-r1                                                                                                        
 * python3_6: running distutils-r1_run_phase python_test                                                                                           
python3.6 setup.py test --verbose                                                                                                                  
running test                                                                                                                                       
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent o
f test runner are encouraged to use tox.                         
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendo
r.urllib3.connection.VerifiedHTTPSConnection object at 0x7f2437c88908>: Failed to establish a new connection: [Errno 101] Network is unreachable',)
': /simple/yapf/                                                         
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendo
r.urllib3.connection.VerifiedHTTPSConnection object at 0x7f2437c88630>: Failed to establish a new connection: [Errno 101] Network is unreachable',)
': /simple/yapf/                                                                                                                                   
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendo
r.urllib3.connection.VerifiedHTTPSConnection object at 0x7f2437d68748>: Failed to establish a new connection: [Errno 101] Network is unreachable',)
': /simple/yapf/                                                                                                                                   
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendo
r.urllib3.connection.VerifiedHTTPSConnection object at 0x7f2437d685f8>: Failed to establish a new connection: [Errno 101] Network is unreachable',)
': /simple/yapf/                    
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendo
r.urllib3.connection.VerifiedHTTPSConnection object at 0x7f2437d68438>: Failed to establish a new connection: [Errno 101] Network is unreachable',)
': /simple/yapf/                                                                                                                                   
ERROR: Could not find a version that satisfies the requirement yapf (from versions: none)
ERROR: No matching distribution found for yapf

**Comment by WPettersson** [comment]
My bad, I didn't realise those had to be added. I followed the advice on https://dev.gentoo.org/~mgorny/python-guide/distutils.html#undesirable-test-dependencies to patch out the coverage/linting requirements (the tests all pass without these extra dependencies, they're only pulled in for the CI in logfury). The only test requirements are now testfixtures and nose, and all tests pass on python 3.6/3.7/3.8.

I also patch out the reference to funcsigs in requirements.txt, since we (I?) only package for python3.6 and above, so it's no longer needed.

**Comment by mgorny** [comment]
Thanks!

**Comment by mgorny** [comment]
For bonus points, you can submit an upstream patch to add `;python_version<"..."` to funcsigs dep.

**Pull request closed without merging**